lib/streamlit/runtime/websocket_session_manager.py

```diff
@@ -0,0 +1,97 @@
+# Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Callable, Dict, List, Optional
+
+from typing_extensions import Final
+
+from streamlit.logger import get_logger
+from streamlit.runtime.app_session import AppSession
+from streamlit.runtime.session_data import SessionData
+from streamlit.runtime.session_manager import (
+    SessionClient,
+    SessionInfo,
+    SessionManager,
+    SessionStorage,
+)
+from streamlit.runtime.uploaded_file_manager import UploadedFileManager
+from streamlit.watcher import LocalSourcesWatcher
+
+LOGGER: Final = get_logger(__name__)
+
+
+class WebsocketSessionManager(SessionManager):
```

Comment:
I wonder if we should consider using a different name here, since this implementation has (so far at least) nothing to do with websockets. If that will continue to be true, what about something like `DefaultSessionManager` or `BasicSessionManager`?

(I don't feel strongly about this, so if there's a good reason to keep the name then let's keep it!)

Reply:
Ack -- I'll add reconsidering the name for this to my list of things to do before the feature branch can get merged into `develop`.

I'm not entirely sure how obviously the implementation will be tied to a websocket connection once the actual reconnect improvements get added, but if it's still the case that there's no clear websocket tie-in after those are done, then I think renaming this to `DefaultSessionManager` would work